model/states.go

```diff
@@ -204,73 +203,66 @@ func (s *State) UnmarshalJSON(data []byte) error {
 			return err
 		}
 		s.DelayState = state
-		return nil
 
 	case string(StateTypeEvent):
 		state := &EventState{}
 		if err := json.Unmarshal(data, state); err != nil {
 			return err
 		}
 		s.EventState = state
-		return nil
 
 	case string(StateTypeOperation):
 		state := &OperationState{}
 		if err := json.Unmarshal(data, state); err != nil {
 			return err
 		}
 		s.OperationState = state
-		return nil
 
 	case string(StateTypeParallel):
 		state := &ParallelState{}
 		if err := json.Unmarshal(data, state); err != nil {
 			return err
 		}
 		s.ParallelState = state
-		return nil
 
 	case string(StateTypeSwitch):
 		state := &SwitchState{}
 		if err := json.Unmarshal(data, state); err != nil {
 			return err
 		}
 		s.SwitchState = state
-		return nil
 
 	case string(StateTypeForEach):
 		state := &ForEachState{}
 		if err := json.Unmarshal(data, state); err != nil {
 			return err
 		}
 		s.ForEachState = state
-		return nil
 
 	case string(StateTypeInject):
 		state := &InjectState{}
 		if err := json.Unmarshal(data, state); err != nil {
 			return err
 		}
 		s.InjectState = state
-		return nil
 
 	case string(StateTypeCallback):
 		state := &CallbackState{}
 		if err := json.Unmarshal(data, state); err != nil {
 			return err
 		}
 		s.CallbackState = state
-		return nil
 
 	case string(StateTypeSleep):
 		state := &SleepState{}
 		if err := json.Unmarshal(data, state); err != nil {
 			return err
 		}
 		s.SleepState = state
-		return nil
-
+	case nil:
```

Comment:
wouldn't it break the unmarshal?
for every state type unmarshaled, the function needs to return back to the execution without error.

Reply:
It's true for other languages, golang doesn't need a return/break in switch.

Reply:
Not break. It's more of an error treatment message if the state does not define the type on json.

Note: I removed the "returns" in the switch cases golang don't need it.